context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,29 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+  const protectedRows = useDatasetsAccessLevel(selectedRows.size > 0 ? [...selectedRows] : []).datasets;
+
+  const containsProtectedDataset = protectedRows.length > 0;
+
+  const errorMessages = [];
+
+  if (selectedRows.size > 10) {
+    errorMessages.push(
+      `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`,
+    );
+  }
+
+  if (containsProtectedDataset) {
+    errorMessages.push(
+      'You have selected protected datasets. Workspaces currently only supports published public datasets. Selected protected datasets are shown below.',
+    );
+  }
```

Comment:
@NickAkhmetov  I agree with your comment during our group call, and this was a good catch. Even though there will be a second version for this, we should still keep the messaging consistent with the current version to uphold the standard and account for potential developmental edge cases (we may not even get to version 2 or forget to change the messaging). I am commenting out V2 messaging and including this temporary one for now.